app/controllers/api/v1/search_controller.rb

```diff
@@ -674,14 +674,20 @@ def self.get_tdr_results(selected_facets:, terms:)
         query_json = ApplicationController.data_repo_client.merge_query_json(facet_query: facet_json, term_query: term_json)
         logger.info "Executing TDR query with: #{query_json}"
         raw_tdr_results = ApplicationController.data_repo_client.query_snapshot_indexes(query_json)
+        added_file_ids = {}
         raw_tdr_results['result'].each do |result_row|
-          results = process_tdr_result_row(result_row, results, selected_facets: selected_facets, terms: terms)
+          results = process_tdr_result_row(result_row, results,
```

Comment:
Rails/Ruby questions:
1. How come `result_row` didn't have to be passed as `row: result_row`?
2. Why can `results` be passed through with no assignment but the other args with the same name as their parameters need to be assigned? ie `results` vs `selected_facets: selected_facets`

Reply:
This is the difference between positional and keyword arguments in Ruby.  Since the private method `process_tdr_result_row` is declared like this:
```
def self.process_tdr_result_row(row, results, selected_facets:, terms:, added_file_ids:)
```

The first two arguments are positional (`row` and `results`), so the order in which they are passed determines their assignment.  The rest are keyword-based, so we need to declare to the method which is which.  I think the confusing part here is that we've mirrored the names of both the local variables and the arguments.  It probably would have been better to use different names for one or the other so that the difference is clearer.

As far as `results` is concerned, it is initialized as an empty hash on the first line of `get_tdr_results`.  After that, we're passing it in to `process_tdr_result_row` and then the rows get merged into that single hash, which is then returned on the last line of the method (hence the assignment of `results = process_tdr_result_row(result_row, results, ...)`.  It all works because of this line:

```
results[short_name] ||= { ... assigning values here }
```

The `||=` operator is Ruby's conditional assignment.  It will assign the following value to the variable, but only if `results[short_name]` evaluates to `nil` or `false`.  Otherwise it will exit silently and leave whatever was already there untouched.  The first time through it initializes the structure, and after that it's a no-op until we get a different value for `short_name`.  This lets us modify it in place without any messy `if/else` logic, or calling `Hash#merge!` repeatedly.

Reply:
Really helpful info, thank you!